File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/base/ApiEndpoints.java
package com.lattis.ellipse.data.network.base;

public enum ApiEndpoints {

    STAGING("Staging", "http://lattisapp-development.lattisapi.io/api/"),
    PRODUCTION("Production", "https://lattisappv2.lattisapi.io/api/");

    public final String name;
    public final String url;

    ApiEndpoints(String name, String url) {
        this.name = name;
        this.url = url;
    }

    @Override
    public String toString() {
        return name;
    }

    public static ApiEndpoints from(String endpoint) {
        for (ApiEndpoints value : values()) {
            if (value.url != null && value.url.equals(endpoint)) {
                return value;
            }
        }
        return null;
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/base/ApiEndpoints.kt
package com.lattis.lattis.data.network.base

enum class ApiEndpoints private constructor(val serverName: String, val url: String) {

    STAGING("Staging", "http://lattisapp-development.lattisapi.io/"),
    PRODUCTION("Production", "https://lattisappv2.lattisapi.io/");

    override fun toString(): String {
        return serverName
    }

    companion object {

        fun from(endpoint: String): ApiEndpoints? {
            for (value in values()) {
                if (value.url != null && value.url == endpoint) {
                    return value
                }
            }
            return null
        }
    }
}
